server/bleep/src/webserver/answer.rs

```diff
@@ -748,12 +748,11 @@ impl Conversation {
                 .ok_or(anyhow!("failed to parse `answer` response, expected array"))?;
 
             let array_of_arrays = json_array
+                .clone()
                 .into_iter()
                 .map(as_array)
                 .collect::<Option<Vec<Vec<_>>>>()
-                .ok_or(anyhow!(
-                    "failed to parse `answer` response, expected array of arrays"
-                ))?;
+                .unwrap_or_else(|| vec![json_array]);
```

Comment:
What error will we get if the model generates a string, so we end up with a `Vec<String>` here?

Reply:
this scenario is permitted (it may recover as more tokens are streamed), however there can be other deviations from schema that are unrecoverable, added logging for those situations.